assets/js/vue-apps/grant-filters.js

```diff
@@ -0,0 +1,93 @@
+import $ from 'jquery';
+import Vue from 'vue';
+
+function init() {
+    const mountEl = document.getElementById('js-grant-filters');
+    if (!mountEl) {
+        return;
+    }
+    new Vue({
+        el: mountEl,
+        delimiters: ['<%', '%>'],
+        data() {
+            let facets = {};
+            let defaultFilters = {
+                amount: '',
+                year: '',
+                programme: '',
+                country: '',
+                localAuthority: ''
+            };
+            let filters = Object.assign({}, defaultFilters);
+            return {
+                facets,
+                defaultFilters,
+                filters
+            };
+        },
+        mounted: function() {
+            // Enable inputs (they're disabled by default to avoid double inputs for non-JS users)
+            $(this.$el)
+                .find('[disabled]')
+                .removeAttr('disabled');
+            // Populate the facets object
+            let facets = this.$el.getAttribute('data-facets');
+            if (facets) {
+                this.facets = JSON.parse(facets);
+            }
+            // Populate any existing query parameters
+            let queryParams = this.$el.getAttribute('data-query-params');
+            if (queryParams) {
+                this.filters = JSON.parse(queryParams);
+            }
+        },
+        methods: {
+            sort: function(sortKey, direction) {
+                this.filters.sort = `${sortKey}|${direction}`;
+                this.filterResults();
+            },
+            filtersToString: function() {
+                return Object.keys(this.filters)
+                    .map(key => {
+                        return `${encodeURIComponent(key)}=${encodeURIComponent(this.filters[key])}`;
+                    })
+                    .join('&');
+            },
+            clearFilters: function() {
+                this.filters = Object.assign({}, this.defaultFilters);
+                this.filterResults();
+            },
+            filterResults: function() {
+                const $form = $(this.$el);
+                const url = $form.attr('action');
+                const $results = $('#js-grant-results');
+                $results.text('Please wait, updating results...');
+                $.ajax({
+                    url: url,
+                    type: 'POST',
```

Comment:
Looking at this now, I'd be strongly tempted to make the action here a separate GET endpoint for a couple of reasons.

- The regular/noscript version of the form doesn't use a post, it's a get action
- The post action makes client-side queries un-cacheable. Using a get endpoint makes sure these queries can be cached

Reply:
If we really wanted to keep the same express handler / endpoint for both then http://expressjs.com/en/4x/api.html#res.format might help with this.

Reply:
This is done – think we should be okay using the same URL for both endpoints as we vary on `Accepts` in Cloudfront. We can test this though.